addons/binding/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/handler/EqualStateFilterHandler.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.astro.handler;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Updates only changed channel states.
+ * 
+ * @author Gerhard Riegler - Initial contribution
+ */
+public abstract class EqualStateFilterHandler extends BaseThingHandler {
```

Comment:
Why do we need this class, it looks rather awkward. In general, state updates are not expensive, so whenever there is a valid value, it can and should be sent, independent whether it has changed or not. If there is really the need to reduce the number of events, this should imho be taken care of within the framework, but not specifically by every binding. So unless you have some strong argument for this class, I would suggest to remove it.